contracts/src/GatewayProxy.sol

```diff
@@ -37,9 +37,6 @@ contract GatewayProxy is IInitializable {
         }
     }
 
-    // Prevent users from unwittingly sending ether to the gateway, as these funds
-    // would otherwise be lost forever.
-    receive() external payable {
-        revert NativeCurrencyNotAccepted();
-    }
```

Comment:
We can't revert this. `GatewayProxy` is immutable forever. Instead you'll need to find a workaround, such as creating an explicit function such as `depositEther` to receive ether.

Reply:
I am not sure I understand this? Surely this is exactly what we want, to use the immutable GatewayProxy address as the storage for the Ether. Should it not be the Gateway implementation that we want to reject `receive` on?

Reply:
Ah I get it, it is already created and cannot be changed in mainnet. Sorry ignore above comment.

Reply:
addressed in [a9768e1](https://github.com/Snowfork/snowbridge/pull/1354/commits/a9768e1562ebaa3eec223e878c7c869a4bf0bd91)